src/Assignments/ProblemIframe.tsx

```diff
@@ -67,8 +69,36 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         // when problem changes, reset lastsubmitted and lastsaved
         setLastSubmittedAt?.(null);
         setLastSavedAt?.(null);
+        setLastSubmission({});
+        updateSubmitActive();
     }, [problem.id]);
 
+    const updateSubmitActive = _.throttle(() => {
+        const submitButtons = iframeRef.current?.contentWindow?.document.getElementsByName('submitAnswers') as NodeListOf<HTMLButtonElement>;
+        const problemForm = iframeRef.current?.contentWindow?.document.getElementById('problemMainForm') as HTMLFormElement;
+        // shouldn't happen - called only onLoad or after interaction with already loaded srcdoc
+        // console.error()?
+        if (_.isNil(submitButtons) || _.isNil(problemForm)) { return; }
+
+        const isClean = _.isEqual(formDataToObject(new FormData(problemForm)), lastSubmission);
+
+        submitButtons.forEach((button: HTMLButtonElement) => {
+            if (isClean) {
+                button.disabled = true;
+                // invisibly stash the button's label (in case there are multiple submit buttons)
+                button.textContent = button.value;
+                button.value = 'Submitted';
+            } else {
+                button.removeAttribute('disabled');
+                if (!_.isNil(button.textContent) && button.textContent != '') {
+                    // put it back and clear the stash - just in case
+                    button.value = button.textContent;
+                    button.textContent = '';
+                }
+            }
+        });
+    }, 1000, {leading:true, trailing:true});
```

Comment:
1 second seems a bit long, I would think it should be something more like 100ms to 500ms